yellowbrick/text/postag.py

```diff
@@ -167,21 +170,46 @@ def __init__(
         self.colormap = colormap
         self.colors = colors
         self.stack = stack
+        self.parser = parser
+        self.tagger = tagger
```

Comment:
It appears that the parser value is validated on `__init__` using the setter, however, the tagger is not validated until `fit()`. It would be my preference if all arguments are validated at construction time rather than downstream because if the visualizer is in a pipeline, there might be a lot of computation that fails before getting to the validation exception. 

Since the `tagger` argument is only used for `nltk`, could I propose the following three options for the `parser` argument: `["nltk", "nltk_wordpunct", "spacy"]` or potentially `["word", "wordpunct", "spacy"]`. This would make validation simpler, earlier, and reduce the number of parameters which I think could be valuable.

Reply:
This is a very good point. However the `self.tagger` argument is used for `spacy` as shown on L299-305

```
spacy = importlib.import_module('spacy')
try:
    if not self.tagger or self.tagger == "en_core_web_sm":
    nlp = spacy.load("en_core_web_sm")
else:
    nlp = spacy.load(self.tagger)
except OSError:
    raise OSError(f"Spacy model '{self.tagger}' has not been downloaded into this environment.")
```
I think that reducing the number of arguments is definitely the most valuable in terms of accessibility of the class. In light of this the solution that I propose is to merge the `tagger` and `parser` arguments into a single argument `parser`, where arguments take the form `"parser__tagger"`. For example `"nltk__wordpunct"`, or `spacy__en_core_web_sm`. The double underscore is for easy string separation if taggers contain single underscores e.g in `spacy`. Let me know what you think of this @bbengfort !

Reply:
I have implemented a solution merging `parser` and `tagger` into the `parser` argument (e.g `nltk_wordpunct` as described above in commit edfdbcf . Looking forward to your thoughts!

Reply:
I very much like the idea of merging the parser and tagger into a single argument - it simplifies things considerably and allows multiple levels of validation in the setter like you've added! Very nice work!